### DISCUSSION TITLE
Show names of places in local language

### DISCUSSION THREAD

**Discussion by komape** [organicmaps/organicmaps#2286] [Ideas]
Places in countries with different names in history or other languages are not displayed with the most local and current one. For example, Bratislava in Slovakia is shown as 'Pressburg" for me as my phone's language is German. According to [Wikipedia](https://en.wikipedia.org/wiki/Bratislava) 'Pressburg' is the old German name for Bratislava but it's not used anymore (at least in my bubble). 

With German there are many more examples in Poland where many places have old German names. This is quite annoying if you travel there and the map shows you different names than the signs on the street.

IMHO it should be configurable if Organic Maps shows me the translated or origin name. What do you think about it?

**Comment by biodranik** [comment]
Please show some screenshots and share links to OSM features, it would be easier to understand the bug.

**Comment by komape** [comment]
The property map for Bratislava shows a German name (`name:de`) which is 'Bratislava' and an old German name (`old_name:de`) which is 'Pressburg'.

See the property map here: https://www.openstreetmap.org/relation/3992676

Organic Maps shows me the old German name instead of the current German one. Not just for Bratislava but many (maybe all) places that have an old German name. IMHO this should be changed. Unfortunately, I have zero experience with C++. Otherwise I would give it a try myself.

![image](https://user-images.githubusercontent.com/25869589/162446490-617f2ec2-b6d3-49b5-8ea1-93d15b82f5f2.png)

**Reply by biodranik** [reply]
What is your device's language(s)/preferred languages list?

**Reply by biodranik** [reply]
Can you please create an issue with the detailed description of this bug? @vng looks like it's a bug in NamesExtractor, there is no unit test for it.

**Comment by vng** [comment]
There is an error in data. See name:de of this node: https://www.openstreetmap.org/node/530544342 that is admin_centre of the relation above. Node's data is more preferable than relation's one for the places.

**Reply by komape** [reply]
Still feels wrong. [openstreetmap.de](https://www.openstreetmap.de/karte/) shows both names, the old and new one (see first image). If I zoom more in it shows only the new one (see second image). I still think it should be changed to the new one.

![image](https://user-images.githubusercontent.com/25869589/162793112-b13a78ae-799d-4b03-b350-a0e81447ba86.png)
![image](https://user-images.githubusercontent.com/25869589/162793153-c5c761a9-5497-4924-b90f-31e11d86d5e2.png)

**Reply by biodranik** [reply]
@komape if you fix [this](https://www.openstreetmap.org/node/530544342) (invalid, right?) `de` translation, OM will draw a proper name after the next maps generation.

**Comment by Jdr-bn** [comment]
First of all: Let me thank all of you guys for the work on Organic Maps: It is a great and helpful application!

I would like to come back to the topic described in this topic and would like to add a suggestion for a feature.
I think this issue is of specific relevance for German-speaking users, but might also relevant in other border regions.

For the following discussion I will use German as an example
We have different name tags at objects in open street maps: "name", showing the name in local language, "name:de", which should represent a "common" translation of the name (e.g. Rome instead of Roma)  in a foreign language (here: the device's language, German) and we have "old_name:de", which should represent historic foreign names / translations, that are not in common use in the foreign language.
Generally, the maps generated from OSM data can be rendered with any of these tags.
In my device, the "leading" name used in the map generated by Organic Maps is the name:de, whilst the name stored in name is also part of the map. They are shown above each other (name:de first) and when zoomed out, only name:de is shown.
![signal-2023-08-22-105602_002](https://github.com/organicmaps/organicmaps/assets/63667433/665139a9-c875-4c5c-97c2-d05d5240f014)
![signal-2023-08-22-105602_003](https://github.com/organicmaps/organicmaps/assets/63667433/0784ff9e-6e01-4a6d-9ff0-7d7807336c91)


When traveling abroad, these translated names stored in name:de, often phonetically totally different from the local language (see: Banska Bytrica vs. Neusohl) and not really useful for orientation / navigation, since they are regularly not visible in the respective countries (e.g. local maps, signs, name of trainstations etc.).
I suggest two options for changes for this issue:
- Change the "order" of "name" and "name:de" when rendering maps with Organic Maps. That is: When zooming out, only the name tag is shown, and the name tag is showing above the name:de tag when zooming in.
- Allow the users to decide to render the maps with "name" tag only somewhere in the settings. Similar to the toggle bar "translitaration to latin letters".

_________________________________________
With my suggestion I'm trying to avoid any discussion about whether it is meaningful / helpful, how these name tag is populated and whether these names should not be better be stored in name:de.
I am convinced: In the example I am dealing with (i.e.: German Names for nearly all cities / villages in those part of eastern Europe, that where part of Habsburg Austria before WW1 or that where  occupied by Nazi Germany during WW2) these names should generally not be part of the name:de tag at all and should be moved to old_name:de). However, unsurprisingly, this is a highly political question which is regularly discussed in the OSM communities, with does not find a clear stance on this (since - of course - there is not only black and white and you could discuss each and every city seperately). Additionally, there seems to be a very active community adding all these names to the names:de tag in Poland, Hungary, Romania, Slovakia, Czech Republic, Estonia, Latvia, Lithuania and Slowenia.

This is something which will be not solved on a short notice and, since the problem of "I want to be able to find the city the train is going to in Organic Maps" is completely unrelated to the question whether the foreign language name tag is populated correctly, i would find it quite handy if users could be able to use the local language name tag more prominently on their phones.


Thank's a lot for the attention :)

**Reply by biodranik** [reply]
The search should correctly find the destination name in any language, right? Does it work in your case?

If the search works, then the issue is only about the map browsing "by using eyes", right? What are the consequences for you of seeing unfamiliar de translations in other countries? Can you still plan your trip by browsing the map? It would be great to understand it in detail before working on solutions.

**Reply by Jdr-bn** [reply]
First of all: Yes: Search works correctly!

Your question got me thinking: Actually it turns out I am maybe a uncommon user of digital maps, since I use the search function nearly never.
Instead I completely use the map by browsing.
When I am at a place and want to go to an other place, I browse to the first place, set "start destination", then browse to the second place, set "end destination" and see the route he is showing me. I am doing that, because I actually try not to be fully dependent on my mobile, and navigating the map gives me a good overview about where things are. Often I am not even looking on the proposed routes, but take routes I see that are possible on the map. Therefor, again, I do a lot of scrolling on the map.

What hinders me with the unfamiliar de translations:
I think most important: The de translation may or not be common as German names for these cities (don't want to judge this). What I can say is: They are completely absent from every-day live in Slovakia. People call the cities by their Slovakian names, road signs, train stations, addresses etc. only show the Slovakian names.
a) Having these de translations shown exclusively on the higher zoom-levels (see first picture above) makes it quite difficult to navigate from one city to an other. Imagine I am in Slovakia, and I know that there is a train from the place I am at, that goes to Banska Bystrica. I want to check out Banska Bystrica on the map, to decide whether it is worth going there. It is not there, but I know the rough direction and the size of the city, so I do some zooming in and out to the level of zoom where the Slovakian names of the city are shown to find the city.
When I have found the city and am quite sure that I want to go there, I set a "Favourites" Pin. If I have not done this and want to look there again later, I search again on the map, often remembering the rough direction or even having learned the weird German name for the city (Neusohl), that I will never use again in my live ;)
Of course, I could use the search. Besides the fact that I somehow just seldom do this, when in foreign countries, the names of the cities are often very weird, and - when not having them available in written form - I tend to forget them. So when someone *told* me about Banska Bystrica and I know the rough location in the country and the approximate size, I will be able to find it on the map by browsing, but I will most likely not be able to spell it that well that the search would find it.

b) When at a place and navigating the area around it (see second picture above) the Slovakian names are shown below the weird de translation. Very nice: However, the de names are completely useless for me. It does not hinder me in using the map, but it is constantly annoying because I have to try not to read them. Since they are above the Slovakian names and in a slightly larger font size, my brain needs to constantly ignore the information that is presented more prominently.

TLDR: As I said in my first post: It does not hinder me using OM (because I want to use offline maps, and OM is the best ;)) so I am not mad if you set different priorities. I can find workarounds and it does, since the local names *are* shown on the map, not make the map unusable in such areas. 
But it is reeeeeeaaaaly annoying and when I find an offline map that only shows local names than I will download that one on my device at well and will use it when travelling eastern Europe ;)

**Comment by vng** [comment]
> (here: the device's language, German)

This is the key point. If the device's language is German, the primary name to show is name:de. And it works not for German only, but for all languages, and it is logical.
I have English language on a device and got this:
<img width="403" alt="Screenshot 2023-08-22 at 11 14 04" src="https://github.com/organicmaps/organicmaps/assets/175612/f60bf902-83cc-448e-a20a-579bf7a5100e">

If the OSM object doesn't have localized name for the device's language, the app shows default "name" (ok, the inner logic is more complicated, but doesn't matter here).

**Comment by vng** [comment]
The mis-variety of German names is a pain, but it should be resolved in OSM, rather than selectively _ban_ some languages in the app.

**Reply by Jdr-bn** [reply]
Thank's for the fast response :)

I understand that for users having English langue devices, the situation is not as complicated in these eastern European countries. However, I would expect that these problems exist in other languages too. There will always be names in the device's local language stored in the OSM data that are phonetically different from the names used in the repective local language.

Although these names (especially when "correctly" tagged as "common name in the devices language" in the OSM data) are meaningful, they might hinder users in easy and handy application of Organic Map when traveling abroad.

Therefore, my proposal is less to ban the language in the app, but to allow the user to focus on the local variants of the map.

However, I of course fully understand if you, or others, deem this either a bad idea, or not worth priorizing.
It was just something which was very present in my user experience these last weeks ;)

**Comment by vng** [comment]
Well, we already have "Transliteration settings", so actually, I don't mind transforming it to some selector like:
- prefer localized name
- prefer default (native) name
- transliterate

Transliteration is close connected to this question, also.

@biodranik @pastk WDYT?

**Reply by muralito** [reply]
When you are "on the road" it seems that <name> tag should be the most useful choice, because it should be what you see in the signs on the street (as the OSM wiki name tag definition: ```
The common default name. Notes:

    For disputed areas, please use the name as displayed on, e.g., street signs for the name tag
    Put all alternatives into either localized name tags (e.g., name:tr/name:el) or the variants (e.g., loc_name/old_name/alt_name)
    Do not abbreviate words: [abbreviations](https://wiki.openstreetmap.org/wiki/Abbreviations)

But maybe when you are browsing the map, or planning a trip you prefer your device language.
```

**Reply by muralito** [reply]
Or maybe some people prefer the app UI language in english (almost always the howtos and error messages are in english forums), and the map data in their native language. If you can, just set a good default, and give the users the choice to change the settings.

**Reply by pastk** [reply]
A big downside for this "a selector in the settings" approach is that one might want to change it when looking at a map of a different country (e.g. I browse between countries maps often) so going to the settings every time will be tedious.

There is a related idea to set a default map language separately from the device language, see https://github.com/organicmaps/organicmaps/issues/391
E.g. in this specific case a user might want to set map language to English while the UI will be in German still. Still, when browsing german-speaking countries, it'll make the English name primary and German one secondary, which is suboptimal...
Also following logic is possible: use a user-set map language as primary unless country's native language matches the device-set lang (then use the device language as primary).

**Reply by muralito** [reply]
yes, but this selector does not have to be only in the settings menu. e.g. it could also be in the "layers" button in the map display, so you can change with a couple of clicks.
